compat/os/mswindows.h

```diff
@@ -107,7 +107,16 @@ SQUIDCEXTERN int WIN32_truncate(const char *pathname, off_t length);
 #define mkdir(p,F) mkdir((p))
 #define pclose _pclose
 #define popen _popen
-#define putenv _putenv
+
+inline int
+setenv(const char * const name, const char * const value, const int overwrite)
+{
+    if (!overwrite && getenv(name))
+        return 0;
+    // overwrite requested or the variable is not set
+    return (_putenv_s(name, value) == 0 ? 0 : -1);
```

Comment:
[AFAICT](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/putenv-s-wputenv-s?view=msvc-170), If value is an empty string, then `_putenv_s()` deletes the named variable. POSIX.1 setenv() does not have that special handling, right? If this is true, our attempt to provide a setenv() replacement has failed. Can you think of a solution other than replacing setenv() calls with some Squid-specific wrapper that returns an error for empty values?

Reply:
You are right that the behavior on these two platforms is different. I created a Windows [test](https://godbolt.org/z/96xxeejsP) showing that _putenv_s() does not create empty variables. Also I checked locally that setenv()/getenv() pair creates/retrieves empty variables. I added an XXX about the difference. As we discussed, we can justify this change by the fact that _putenv() worked similarly in this respect.

Reply:
BTW, it looks like that Windows folks found it inconvenient to be unable to create empty variables, as they recently [added](https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/9.0/empty-env-variable) this support to .NET platform.